docs/user/forms.md

```diff
@@ -18,11 +18,7 @@ from pypdf import PdfReader, PdfWriter
 reader = PdfReader("form.pdf")
 writer = PdfWriter()
 
-page = reader.pages[0]
-fields = reader.get_fields()
-
-writer.add_page(page)
-
+writer.append_pages_from_reader(reader)
```

Comment:
I think that using `append()`may be better to be used

Reply:
@pubpub-zz I have tried `append` and provided test cases to reproduce the bug in the past (April 13th, see above).   Please take over this PR.

Reply:
oups missed it.
The document you have provided uses some advance features : Some extra information which are not page dependent but stored in the root object. no page function, even `append()` can not copy properly the information. 
However the information are available ; the following code:
```
import pypdf
w=pypdf.PdfWriter()
w.append("f941filled.pdf")
w.write("test.pdf")
```
provides the following result:
![image](https://github.com/py-pdf/pypdf/assets/4083478/3852fd6b-fef3-4f64-8580-9a032d8278de)

If you want to duplicate more properly you should use `PdfWriter(clone_from=...)` or `.clone_document_from_reader(...)`
the code:
```
import pypdf
w=pypdf.PdfWriter(clone_from = "f941filled.pdf")
w.write("test.pdf")
```
However as the document has some internal signature, an warning is reported
![image](https://github.com/py-pdf/pypdf/assets/4083478/b43b7d04-56ff-4e59-b8bc-60f21c1e5293)
The document is still readable with the fields filled in but you can not modify it anymore (locked)
![image](https://github.com/py-pdf/pypdf/assets/4083478/8763ef5b-6a52-433c-86f5-73ecca347aca)

When you open the documents with pdf.JS, both can be modified

Reply:
I've tested your example with `append_pages_from_reader()` but this solution breaks the form which is stored outside of the pages : `gef_fields()` and others can not work in these conditions